swarm/api/api.go

```diff
@@ -336,18 +391,23 @@ func (self *Api) AppendFile(mhash, path, fname string, existingSize int64, conte
 }
 
 func (self *Api) BuildDirectoryTree(mhash string, nameresolver bool) (key storage.Key, manifestEntryMap map[string]*manifestTrieEntry, err error) {
+	apiBuildDirTreeCount.Inc(1)
```

Comment:
Do you think this is a useful metric? I'm not so sure.

Reply:
We've been focusing on setting up the library, the reporter and the infrastructure around metrics. I agree that not all of the measurements currently are useful and meaningful - there will definitely be iterations on them. At the same time I am happy to fix those that are immediately known to be not useful in this PR. :)

Reply:
I have been developing mostly web REST APIs in previous years. Each such a REST API usually is there because it serves a specific function and a user requirement/use case. From this point of view, I interpreted this function something like a public API function - in which case we may be interested to know how many times each of these are being called. If my interpretation actually is incorrect, in that this API call isn't very relevant, is maybe semantically not a user-facing API endpoint, or is maybe even obsolete/deprecated, I happy for it to be removed

Reply:
Given you are questioning whether this is useful, we should just drop it unless we decide it is useful. I personally don't think it is a useful metric.

Reply:
I was not really questioning the usefulness of the metric, but rather asking for clarification about the role or, maybe the relevance of the API itself in terms of (public) use case. I will proceed to remove it, but I would also welcome better documentation on such supposedly API functions, in order to understand why they are there and their relevance for users of the API.